common/shared.c

```diff
@@ -1062,7 +1072,7 @@ int ft_exchange_keys(struct fi_rma_iov *peer_iov)
 		} else {
 			rma_iov->addr = 0;
 		}
-		rma_iov->key = fi_mr_key(mr);
+		rma_iov->key = mr_key;
```

Comment:
Registration for remote access is required.  fi_mr_key() should be usable in all cases.

Reply:
I've removed this change. Now fi_mr_key() is used in all cases